src/main/java/org/entando/kubernetes/service/digitalexchange/BundleUtilities.java

```diff
@@ -217,10 +226,33 @@ public static String truncatePodPrefixName(String podPrefixName) {
         return podPrefixName;
     }
 
-    private static String composeIngressPathFromDockerImage(DockerImage image) {
-        return "/" + String.join("/",
-                makeKubernetesCompatible(image.getOrganization()),
-                makeKubernetesCompatible(image.getName()));
+    private static String composeIngressPathFromDockerImage(PluginDescriptor descriptor) {
+
+        DockerImage image = descriptor.getDockerImage();
+
+        List<String> ingressSegmentList = new ArrayList<>(Arrays.asList(image.getOrganization(), image.getName()));
+
+        if (BundleUtilities.getPluginDescriptorIntegerVersion(descriptor) < 3) {
+            ingressSegmentList.add(image.getVersion());
+        }
+
+        List<String> kubeCompatiblesSegmentList = ingressSegmentList.stream()
+                .map(BundleUtilities::makeKubernetesCompatible).collect(Collectors.toList());
+
+        return "/" + String.join("/", kubeCompatiblesSegmentList);
+    }
+
+    public static Integer getPluginDescriptorIntegerVersion(PluginDescriptor pluginDescriptor) {
+
+        if (! StringUtils.hasLength(pluginDescriptor.getDescriptorVersion())) {
+            return pluginDescriptor.isVersion1() ? 1 : 2;
+        } else {
+            Matcher matcher = PLUGIN_DESCRIPTOR_VERSION_PATTERN.matcher(pluginDescriptor.getDescriptorVersion());
+            if (!matcher.matches()) {
+                throw new InvalidBundleException("The plugin descriptor version does not match the expected format");
+            }
+            return Integer.parseInt(matcher.group(2));
```

Comment:
@firegloves hande NumberFormatException and rethrow as InvalidBundleException with a proper message

Reply:
@w-caffiero-entando I'm adding it but the check is already made with the regex, don't it?

Reply:
yeah, right, missed it